plugins/woocommerce/src/Internal/Admin/Agentic/AgenticWebhookPayloadBuilder.php

```diff
@@ -0,0 +1,230 @@
+
+		if ( ! empty( $order_received_url ) ) {
+			return $order_received_url;
+		}
+
+		// Fallback to admin edit URL if customer URL not available
+		return admin_url( 'post.php?post=' . $order->get_id() . '&action=edit' );
+	}
+
+	/**
+	 * Map WooCommerce order status to ACP status.
+	 *
+	 * ACP statuses: created, manual_review, confirmed, canceled, shipped, fulfilled
+	 *
+	 * @param string $wc_status WooCommerce order status.
+	 * @return string ACP status.
+	 */
+	private function map_order_status( $wc_status ) {
+		// Remove 'wc-' prefix if present.
+		$wc_status = str_replace( 'wc-', '', $wc_status );
+
+		$status_map = array(
+			// WooCommerce status => ACP status.
+			'pending'        => ACPOrderStatus::CREATED,
+			'processing'     => ACPOrderStatus::CONFIRMED,
+			'on-hold'        => ACPOrderStatus::MANUAL_REVIEW,
+			'completed'      => ACPOrderStatus::FULFILLED,
+			'cancelled'      => ACPOrderStatus::CANCELED,
+			'canceled'       => ACPOrderStatus::CANCELED, // Support both spellings.
+			'refunded'       => ACPOrderStatus::FULFILLED, // Refunded orders are still fulfilled.
+			'failed'         => ACPOrderStatus::CANCELED,
+			'checkout-draft' => ACPOrderStatus::CREATED,
+		);
+
+		/**
+		 * Filter the WooCommerce to ACP order status mapping.
+		 *
+		 * Allows extensions to map custom WooCommerce order statuses to ACP order statuses.
+		 * The mapped status must be one of: created, manual_review, confirmed, canceled, shipped, fulfilled.
+		 *
+		 * @since 10.3.0
+		 *
+		 * @param array  $status_map Associative array of WooCommerce status => ACP status.
+		 * @param string $wc_status  The WooCommerce order status being mapped.
+		 */
+		$status_map = apply_filters( 'woocommerce_agentic_webhook_order_status_map', $status_map, $wc_status );
+
+		// Get mapped status or default to 'created'.
+		$mapped_status = isset( $status_map[ $wc_status ] ) ? $status_map[ $wc_status ] : ACPOrderStatus::CREATED;
+
+		// Validate the mapped status is a valid ACP status.
+		if ( ! ACPOrderStatus::is_valid( $mapped_status ) ) {
+			// Log a warning for invalid status but continue with fallback.
+			// phpcs:ignore WordPress.PHP.DevelopmentFunctions.error_log_error_log
+			error_log(
+				sprintf(
+					'Invalid ACP order status "%s" returned by woocommerce_agentic_webhook_order_status_map filter for WooCommerce status "%s". Using "created" as fallback.',
+					$mapped_status,
+					$wc_status
+				)
+			);
+			return ACPOrderStatus::CREATED;
+		}
+
+		return $mapped_status;
+	}
+
+	/**
+	 * Build refunds data for the order.
+	 *
+	 * @param WC_Order $order Order object.
+	 * @return array Array of refunds.
+	 */
+	private function build_refunds_data( $order ) {
+		$refunds_data = array();
+		$refunds      = $order->get_refunds();
+
+		if ( empty( $refunds ) ) {
+			return $refunds_data;
+		}
+
+		foreach ( $refunds as $refund ) {
+			$refunds_data[] = $this->build_single_refund_data( $refund );
+		}
+
+		return $refunds_data;
```

Comment:
Nitpick mode engaged... 😃

I think this can be way shorter:

```suggestion
		return array_map(
			[ $this, 'build_single_refund_data' ],
			$order->get_refunds()
		);
```